Handle the case where an index template that exists in the state is missing from the ES domain (e.g. has been manually deleted.)

TonyLovesDevOps opened the pull request:
Thanks for authoring this provider; I love managing ES in code with `terraform` instead of in my brain with clicks and keystrokes!

I ran into an issue when I was replacing an ES domain, where if an index template was stored in the TF state but didn't exist within ES, the `terraform plan` would error with a 404 error. This PR fixed it for me locally.

I _thought about_ adding tests, but I'm a n00b to both `go` and `travis`, so I didn't.

TonyLovesDevOps commented:
The CI build failed on a file that I didn't touch. From the [build log](https://travis-ci.com/github/phillbaker/terraform-provider-elasticsearch/jobs/449190296#L351):
```
125.49s$ golangci-lint run --new-from-rev=bf51aaa --timeout=10m
level=warning msg="[runner] Can't process result by diff processor: can't prepare diff by revgrep: could not read git repo: error executing git diff \"bf51aaa\" \"\": exit status 128"
es/resource_elasticsearch_kibana_object.go:223:7: Error return value of `d.Set` is not checked (errcheck)
	d.Set("body", result)
	     ^
The command "golangci-lint run --new-from-rev=bf51aaa --timeout=10m" failed and exited with 1 during .
```

phillbaker commented:
This started failing a couple of days ago on all branches, I’m looking into
it.

For this PR, can you add a line to the change log please?

On Thu, Nov 26, 2020 at 2:15 PM Tony DevOps <notifications@github.com>
wrote:

> The CI build failed on a file that I didn't touch. From the build log
> <https://travis-ci.com/github/phillbaker/terraform-provider-elasticsearch/jobs/449190296#L351>
> :
>
> 125.49s$ golangci-lint run --new-from-rev=bf51aaa --timeout=10m
> level=warning msg="[runner] Can't process result by diff processor: can't prepare diff by revgrep: could not read git repo: error executing git diff \"bf51aaa\" \"\": exit status 128"
> es/resource_elasticsearch_kibana_object.go:223:7: Error return value of `d.Set` is not checked (errcheck)
> 	d.Set("body", result)
> 	     ^
> The command "golangci-lint run --new-from-rev=bf51aaa --timeout=10m" failed and exited with 1 during .
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/phillbaker/terraform-provider-elasticsearch/pull/116#issuecomment-734448940>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAXCKLRLGM7VANXG5FEAU3SR2SNZANCNFSM4UECGMUA>
> .
>

phillbaker commented:
I've updated the CI config to fix the issue (https://github.com/phillbaker/terraform-provider-elasticsearch/commit/afdf0a0adb0cf0569c1d86bc582713d869ca9ea3), can you please rebase off of master?

TonyLovesDevOps commented:
> I've updated the CI config to fix the issue ([afdf0a0](https://github.com/phillbaker/terraform-provider-elasticsearch/commit/afdf0a0adb0cf0569c1d86bc582713d869ca9ea3)), can you please rebase off of master?

Cool, I rebased and added a `CHANGELOG.md` entry.
